### PR TITLE
user details screen, settings screen , localisation

### PR DESCRIPTION
**Pull request by vijay-shuru**
- added user details screen - where user can provide info like name, gender, dob, country, language
- Updated settings screen where user can edit provided details and can change appearance mode between light and dark
- support localisation with English, Hindi, Chinese, French, Spanish, Arabic

### CODE REVIEW

File: SwiftAppTemplate/Dummy-Use&Delete/EditUserDetailsScreen.swift [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+                dismiss()
+            }
+        }
+        .onAppear(){
+            initializeDetails()
+        }
+     
+    }
+    
+    private var nameField : some View {
```
Comment (gunveer-shuru) [inline SwiftAppTemplate/Dummy-Use&Delete/EditUserDetailsScreen.swift]: All these fields like nameField and emailField could be one struct, genderField and countryField could be one struct.

File: SwiftAppTemplate/Dummy-Use&Delete/SettingsScreen.swift [hunk 2]
```diff
@@ -20,8 +23,30 @@ struct SettingsScreen: View {
             VStack(spacing: 30){
                 
                 Header(text: "Settings", hasBackButton: true, onBackArrowClick: {dismiss()})
+
```
Comment (gunveer-shuru) [inline SwiftAppTemplate/Dummy-Use&Delete/SettingsScreen.swift:26]: The below titleValueViews can be put in one struct for better clarity.

File: SwiftAppTemplate/Dummy-Use&Delete/SettingsScreen.swift [hunk 3]
```diff
@@ -20,8 +23,30 @@ struct SettingsScreen: View {
...
+                
+                TitleValueView(title: "DateOfBirth", value: viewModel.dob)
+        
+                TitleValueView(title: "Country", value: viewModel.country)
+
+                TitleValueView(title: "Language", value: viewModel.language)
+
                 
-                customTextFields
+                TextButton(onClick: {
```
Comment (gunveer-shuru) [inline SwiftAppTemplate/Dummy-Use&Delete/SettingsScreen.swift]: This text button can be put in a variable.

File: SwiftAppTemplate/Screens/Authentication/AuthorizationScreen.swift [hunk 4]
```diff
@@ -25,39 +16,27 @@ struct AuthorizationScreen: View {
     var body: some View {
         ZStack {
             VStack {
-                Header(text: getLocalString(title))
-                nameField
-                    .padding(.vertical, 20)
+                Header(text: "Login/SignUp")
```
Comment (gunveer-shuru) [inline SwiftAppTemplate/Screens/Authentication/AuthorizationScreen.swift]: No localized string here in the header?

File: SwiftAppTemplate/Screens/UserDetails/UserDetailsScreen.swift [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+       }
+    var startDate = Calendar.current.date(byAdding: .year, value: -100, to: Date()) ?? Date()
+    let onCompleted: () -> Void
+    
+    
+    var body: some View {
+        VStack {
+            Header(text: "UserDetails")
+            
+            nameField
```
Comment (gunveer-shuru) [inline SwiftAppTemplate/Screens/UserDetails/UserDetailsScreen.swift]: Same kind of fields also got used in edituserdetails screen is there a possibility of extracting them into reuseable structs?

File: SwiftAppTemplate/Screens/UserDetails/UserDetailsScreen.swift [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+                .font(.notoSansMedium16)
+                .foregroundColor(.primaryNavyBlue)
+            CustomTextField(inputText: Binding.constant(selectedLanguage), placeholder: getLocalString("SelectLanguagePlaceHolder"), cornerRadius: 10, borderColor: .primaryNavyBlue)
+                .onTapGesture {
+                    openDeviceSettings()
+                }
+        }
+    }
+    
+    func hasEnteredAllDetails() -> Bool {
```
Comment (gunveer-shuru) [inline SwiftAppTemplate/Screens/UserDetails/UserDetailsScreen.swift:67]: Same for these functions as well, they can be extracted in a common util class perhaps?

**Review by gunveer-shuru — CHANGES_REQUESTED** [review]
I'm thinking why not let's put all the local string keys in one file and then use them where they are required. This would make it easy to edit the keys in the future and reduce the possibility of mistakes. What do you say?

File: SwiftAppTemplate/Screens/Authentication/AuthorizationScreen.swift (hunk 4 above)
Comment (vijay-shuru) [reply SwiftAppTemplate/Screens/Authentication/AuthorizationScreen.swift]: text variable of Header view is type of LocalizedStringKey

File: SwiftAppTemplate/Screens/UserDetails/UserDetailsScreen.swift (hunk 6 above)
Comment (vijay-shuru) [reply SwiftAppTemplate/Screens/UserDetails/UserDetailsScreen.swift:67]: one difference UserDetailsScreen has no email field , but  Edit screen has email field

File: SwiftAppTemplate/Dummy-Use&Delete/EditUserDetailsScreen.swift (hunk 1 above)
Comment (vijay-shuru) [reply SwiftAppTemplate/Dummy-Use&Delete/EditUserDetailsScreen.swift]: fixed, gender and country fields have different Stack view Vstack and Hstack

File: SwiftAppTemplate/Dummy-Use&Delete/SettingsScreen.swift (hunk 2 above)
Comment (vijay-shuru) [reply SwiftAppTemplate/Dummy-Use&Delete/SettingsScreen.swift:26]: fixed

File: SwiftAppTemplate/Dummy-Use&Delete/SettingsScreen.swift (hunk 3 above)
Comment (vijay-shuru) [reply SwiftAppTemplate/Dummy-Use&Delete/SettingsScreen.swift]: fixed

File: SwiftAppTemplate/Screens/UserDetails/UserDetailsScreen.swift (hunk 5 above)
Comment (vijay-shuru) [reply SwiftAppTemplate/Screens/UserDetails/UserDetailsScreen.swift]: fixed

**Review by gunveer-shuru — APPROVED** [review]

File: SwiftAppTemplate/Dummy-Use&Delete/EditUserDetailsScreen.swift [hunk 7]
```diff
@@ -63,6 +63,8 @@ struct EditUserDetailsScreen: View {
                 
                 TextButton(onClick: {
                     if hasEnteredAllDetails() {
+                        AnalyticsManager.logButtonClickEvent(buttonType: .primary, label: "Update")
+                        AnalyticsManager.logCustomEvent(eventType: .updateUser, properties: ["name": name, "email": email, "gender": selectedGender, "country": selectedCountry, "language": selectedLanguage])
```
Comment (gunveer-shuru) [inline SwiftAppTemplate/Dummy-Use&Delete/EditUserDetailsScreen.swift]: User object instead of params

**Review by gunveer-shuru — APPROVED** [review]

File: SwiftAppTemplate/Dummy-Use&Delete/EditUserDetailsScreen.swift (hunk 7 above)
Comment (vijay-shuru) [reply SwiftAppTemplate/Dummy-Use&Delete/EditUserDetailsScreen.swift]: this not supported in analytics events

**Pull request merged**